### DISCUSSION TITLE
CredentialsProvider + Database session

### DISCUSSION THREAD

**Discussion by redimongo** [nextauthjs/next-auth#4457] [Help]
I don't know about others but JWT session while helpful, they can create issues and for applications that need to limit 1 user per session and prevent users from logging in on multiple accounts (ie: banks), it almost makes it impossible without adding extra lines of code which is what I am currently having to do, to add the JWT session inside the session collection inside Mongodb.

I wish there was an easier way, that allowed developers who want to not rely on Auth0 (which is paid) to create a oAuth server quick and easy - but until I find that solution I need away to do this with just username and password.

is there away to store CredentialsProvider JWT session inside the database - even if not officially supported.

```
session: {
    strategy: 'database',
}
```

**Comment by julianmcapicua** [comment]
I need to add the same feature to my app, it would be very helpful to have some orientation on how to do it.
I've been looking at the next-auth source code and I can't manage to figure out how to add credentials login using the database strategy.

@redimongo Were you able to accomplish this?
